feat!: emacs tests

MuntasirSZN opened the pull request.

MuntasirSZN committed:
feat: emacs stuff (wip)

MuntasirSZN committed:
chore: format

MuntasirSZN committed:
chore: format (use elfmt only)

MuntasirSZN committed:
some test stuff

MuntasirSZN committed:
Merge branch 'main' into feat/emacs

MuntasirSZN committed:
feat: add tests!

MuntasirSZN committed:
lockfile

MuntasirSZN committed:
fix

MuntasirSZN committed:
Update emacs-checks.yml

MuntasirSZN committed:
Merge branch 'main' into feat/emacs

Kyure-A commented:
> I don't have enough knowledge to review this PR. @flowerhill @Kyure-A Can you review this PR? I heard that you have knowledge about Emacs Lisp.

I'm a bit busy so I'll review it within a week.

MuntasirSZN committed:
fix

MuntasirSZN committed:
breaking change

MuntasirSZN committed:
Merge branch 'main' into feat/emacs

MuntasirSZN committed:
Merge remote-tracking branch 'origin/main' into feat/emacs

MuntasirSZN committed:
Merge branch 'main' into feat/emacs

MuntasirSZN committed:
refactor(emacs)!: rename cursor api, update hooks, and harden overlays

• Simplify rustowl-cursor to always issue async requests for better responsiveness; remove explicit lsp-mode/workspace guard
• Rework rustowl-cursor-call to build params via plist for clarity and reduce boilerplate
• Rename public autoloads to consistent rustowl-* names and update mode hooks accordingly
• Make rustowl-underline robust by clamping to buffer bounds and handling reversed ranges to avoid overlay errors
• Change add-hook/remove-hook usage to non-buffer-local, aligning with typical mode-level activation

BREAKING CHANGE:

• Function renames:
 • enable-rustowl-cursor -> rustowl-enable-cursor
 • disable-rustowl-cursor -> rustowl-disable-cursor
• Hook behavior changed: post-command-hook is no longer added/removed buffer-locally from these helpers. If you relied on buffer-local hooks, pass a local arg in your own
configuration or adapt your setup accordingly.

MuntasirSZN committed:
fix(emacs): add aliases and buffer‑local hooks; restore LSP guard; optimize line calc

- Make post-command hooks buffer-local to avoid global side effects
- Restore LSP active/workspace check in `rustowl-cursor`
- Optimize line count using `(line-number-at-pos (point-max))`

MuntasirSZN committed:
fix: tests

MuntasirSZN committed:
Merge branch 'main' into feat/emacs

cordx56 commented on .gitignore:
```diff
@@ -16,3 +16,4 @@ man/
 memory-bank/
 security-logs/
 benchmark-summary.*
+.eask/
```
tail newline?

MuntasirSZN committed:
Remove .eask from .gitignore

MuntasirSZN committed:
Fix script paths for Rustowl setup in CI

Updated the script paths for building and installing Rustowl in the workflow.

MuntasirSZN committed:
Update emacs-checks.yml

MuntasirSZN committed:
Merge remote-tracking branch 'upstream/main' into feat/emacs

MuntasirSZN committed:
chore: fixes

MuntasirSZN committed:
use snapshot versions
